### PR TITLE
fix: also install cargo-dist on build-local

### PR DESCRIPTION
**Pull request by ruben-arts**
It worked 🎉  https://github.com/prefix-dev/pixi/releases/tag/v0.39.2-prerelease.2

Changes we had to make:
- We now have a fork of `cargo-dist` at: https://github.com/ruben-arts/cargo-dist/pull/1
	- This changes how the normal binaries are structured for `tar.gz`
	- It adds notarization for the macos binaries
- We unpack the `cargo-dist` generated archives and re-upload the binaries in unpacked form, this is important for `setup-pixi`.
- We don't announce fully be release as draft.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -262,6 +267,61 @@ jobs:
...
+            base_name=$(basename "$artifact" .tar.gz)
+
+            # Create a directory for the specific artifact
+            mkdir -p unpacked-artifacts/$base_name
+
+            # Extract the artifact contents
+            tar -xvf "$artifact" -C unpacked-artifacts/$base_name
+
+            # Move the pixi binary to the base name
+            mv unpacked-artifacts/$base_name/pixi unpacked-artifacts/$base_name
```
Comment (nichmor) [inline .github/workflows/release.yml:285]: I would say that for safety reason maybe we could remove after moving
` unpacked-artifacts/$base_name/pixi` folder? so we accidently don't upload it or something

File: Cargo.toml [hunk 2]
```diff
@@ -362,7 +362,7 @@ tokio = { workspace = true, features = ["rt"] }
 # Config for 'dist'
 [workspace.metadata.dist]
 # The preferred dist version to use in CI (Cargo.toml SemVer syntax)
-cargo-dist-version = "0.26.0-prerelease.3"
+cargo-dist-version = "0.25.1"
```
Comment (nichmor) [inline Cargo.toml:365]: if we downgrade now, do we need to change something else in the configuration?

File: .github/workflows/release.yml (hunk 1 above)
Comment (ruben-arts) [reply .github/workflows/release.yml]: Changed that

**Review by nichmor — APPROVED** [review]
looks good from my side

File: Cargo.toml [hunk 3]
```diff
@@ -165,7 +165,7 @@ license.workspace = true
 name = "pixi"
 readme.workspace = true
 repository.workspace = true
-version = "0.39.1"
+version = "0.39.2-prerelease.0"
```
Comment (ruben-arts) [inline Cargo.toml:168]:
```suggestion
version = "0.39.2-prerelease.1"
```

File: crates/pixi_consts/src/consts.rs [hunk 4]
```diff
@@ -14,7 +14,7 @@ pub const CONFIG_FILE: &str = "config.toml";
 pub const PIXI_DIR: &str = ".pixi";
 pub const PIXI_VERSION: &str = match option_env!("PIXI_VERSION") {
     Some(v) => v,
-    None => "0.39.1",
+    None => "0.39.2",
```
Comment (ruben-arts) [inline crates/pixi_consts/src/consts.rs:17]:
```suggestion
    None => "0.39.1",
```

File: Cargo.toml [hunk 5]
```diff
@@ -165,7 +165,7 @@ license.workspace = true
 name = "pixi"
 readme.workspace = true
 repository.workspace = true
-version = "0.39.1"
+version = "0.39.2-prerelease.2"
```
Comment (ruben-arts) [inline Cargo.toml:168]:
```suggestion
version = "0.39.1"
```

File: Cargo.lock [hunk 6]
```diff
@@ -3590,7 +3590,7 @@ dependencies = [
 
 [[package]]
 name = "pixi"
-version = "0.39.1"
+version = "0.39.2-prerelease.2"
```
Comment (ruben-arts) [inline Cargo.lock:3593]:
```suggestion
version = "0.39.1"
```

**Review by Hofer-Julian — APPROVED** [review]
Let's get a working release again!

**Pull request merged**